File: application/src/main/java/de/pbauerochse/worklogviewer/domain/timerangeprovider/CurrentWeekTimerangeProvider.java
package de.pbauerochse.worklogviewer.domain.timerangeprovider;

import de.pbauerochse.worklogviewer.domain.ReportTimerange;
import de.pbauerochse.worklogviewer.report.TimeRange;

/**
 * @author Patrick Bauerochse
 * @since 13.04.15
 */
public class CurrentWeekTimerangeProvider extends BaseTimerangeProvider {

    @Override
    public TimeRange getTimeRange() {
        return TimeRange.currentWeek();
    }

    @Override
    public ReportTimerange getReportTimerange() {
        return ReportTimerange.THIS_WEEK;
    }
}

File: application/src/main/java/de/pbauerochse/worklogviewer/timerange/CurrentWeekTimerangeProvider.kt
package de.pbauerochse.worklogviewer.timerange

import de.pbauerochse.worklogviewer.timereport.TimeRange
import de.pbauerochse.worklogviewer.util.FormattingUtil.getFormatted
import java.time.LocalDate

/**
 * The current week as [TimerangeProvider]
 */
internal object CurrentWeekTimerangeProvider : TimerangeProvider {
    override val label: String = getFormatted("timerange.thisweek")
    override val settingsKey: String = "THIS_WEEK"
    override val isComputed: Boolean = true
    override fun buildTimeRange(start: LocalDate?, end: LocalDate?): TimeRange {
        return TimeRange.currentWeek()
    }
}
